newsfragments/2989.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed a bug where :class:`trio.testing.RaisesGroup` would check the number of exceptions in the raised `ExceptionGroup` before flattening subgroups, leading to incorrectly failed matches with ``flatten_subgroups=True`` (previously ``strict=False``).
```

Comment:
This note seems odd, given that `flatten_subgroups` was just added in this PR.  I think it should only mention `strict=False`, which is the only way people can be hitting this bug.

Reply:
flipped the references to them